replacement_scan.go

```diff
@@ -0,0 +1,60 @@
+package duckdb
+
+/*
+   #include <stdlib.h>
+   #include <duckdb.h>
+   void replacement_scan_cb(duckdb_replacement_scan_info info, const char *table_name, void *data);
+   typedef const char cchar_t;
+*/
+import "C"
+import (
+	"unsafe"
+)
+
+type ReplacementScanCallback func(tableName string) (string, []any, error)
+
+var replacementScanState struct {
```

Comment:
Is there probably another way to inject the callback into the `replacement_scan_cb` function instead of declaring it on the package level? What happens when opening multiple DBs and registering a `ReplacementScanCallback` for each of them? Will there be an interference between the multiple declarations, because `replacementScanState` will be overwritten?

Reply:
Good point. Didn't think about such a scenario, luckily it was possible to store the callback inside the duckdb instance.